classes/submission/SubmissionDAO.inc.php

```diff
@@ -111,10 +112,104 @@ function _fromRow($row) {
 		$submission->setLanguage($row['language']);
 		$submission->setCommentsToEditor($row['comments_to_ed']);
 
-		$this->getDataObjectSettings('submission_settings', 'submission_id', $submission->getId(), $submission);
+		$this->getSubmissionSettings($submission, $submissionRevision);
 
 		return $submission;
 	}
+	
+	/**
+	 * Get contents of the submission_settings table
+	 * @param $submission Submission object
+	 * @param $submissionRevision int optional
+	 */
+	function getSubmissionSettings($submission, $submissionRevision = null) {
+
+		if ($submissionRevision == null) {
+			$sql = "SELECT * FROM submission_settings WHERE submission_id = ?";
+			$params = array($submission->getId());
+		} else {
+			$sql = "SELECT * FROM submission_settings WHERE submission_id = ? AND submission_revision = ?";
+			$params = array($submission->getId(), $submissionRevision);
+		}
+		$result = $this->retrieve($sql, $params);
+
+		while (!$result->EOF) {
+			$row = $result->getRowAssoc(false);
+
+			$submission->setData(
+				$row['setting_name'],
+				$this->convertFromDB(
+					$row['setting_value'],
+					$row['setting_type']
+				),
+				empty($row['locale']) ? null : $row['locale']
+			);
+
+			$result->MoveNext();
+		}
+		$result->Close();
+	}
+	
+	/**
+	 * Get all revisions for a submission; default settings: 
+	 * 1) Retrieve only the previous revisions, not the current one
+	 * 2) Get only the id's; optionally the title can be retrieved, too
+	 * 3) Default sorting order is ASC
+	 * @param $submissionId int
+	 * @param $showAll boolean
```

Comment:
These parameters could use some inline documentation, e.g.:

    * @param $showAll boolean false to exclude current revision; true to fetch all revisions including current